internal/impl/refresh.go

```diff
@@ -0,0 +1,75 @@
+package impl
+
+import (
+	"fmt"
+	"os"
+	"strings"
+)
+
+func (d *Devbox) IsDirenvActive() bool {
+	return strings.TrimPrefix(os.Getenv("DIRENV_DIR"), "-") == d.projectDir
+}
+
+func (d *Devbox) isRefreshAliasSet() bool {
+	return os.Getenv(d.refreshAliasEnvVar()) == d.refreshCmd()
+}
+
+func (d *Devbox) refreshAliasEnvVar() string {
+	return "DEVBOX_REFRESH_ALIAS_" + d.projectDirHash()
+}
+
+func (d *Devbox) isGlobal() bool {
+	globalPath, _ := GlobalDataPath()
+	return d.projectDir == globalPath
+}
+
+// In some cases (e.g. 2 projects somehow active at the same time),
+// refresh might not match. This is a tiny edge case, so no need to make UX
+// great, we just print out the entire command.
+func (d *Devbox) refreshAliasOrCommand() string {
+	if !d.isRefreshAliasSet() {
+		return d.refreshCmd()
+	}
+	return d.refreshAliasName()
+}
```

Comment:
will this occur if devbox global is enabled in the user's shellrc, and they are now using a local devbox project?

Reply:
If not, could you explain why? I was struggling to figure it out

Reply:
We use different refresh commands `refresh` vs `refresh-global` so no, this will only happen if we ever somehow activate 2 non-global devboxes at the same time (e.g. manually running shellenv might do it). Very edge case. But if we don't do this, we'll tell the user to use `refresh` but that won't work.